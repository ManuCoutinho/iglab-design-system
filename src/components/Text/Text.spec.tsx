import { screen, render } from '@testing-library/react'
import { Text } from '.'

describe('<Text/>', () => {
  it('should render Text correctly ', () => {
    render(<Text>Text</Text>)
    expect(screen.getByText(/Text/i)).toBeInTheDocument()
  })
  it('should render Text with right size', () => {
    render(<Text size='sm'>Text</Text>)
    expect(screen.getByText(/Text/i)).toHaveClass('text-xs')
  })
  it('should render Text with className prop', () => {
    render(<Text className='text-blue-100'>Text</Text>)
    expect(screen.getByText(/Text/i)).toHaveClass('text-blue-100')
  })
  it('match to snapshot', () => {
    const { container } = render(<Text>Text</Text>)
    expect(container.firstChild).toMatchSnapshot()
  })
})
