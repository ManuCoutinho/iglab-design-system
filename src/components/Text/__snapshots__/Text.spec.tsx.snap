// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text/> match to snapshot 1`] = `
<span
  class="text-gray-100 font-sans text-sm"
>
  Text
</span>
`;
